webapp/src/components/Countdown/Countdown.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { t } from '@dapps/modules/translation/utils'
+
+import './Countdown.css'
+
+export default class Countdown extends React.PureComponent {
+  static propTypes = {
+    date: PropTypes.number.isRequired
+  }
+
+  static defaultProps = {
+    date: new Date()
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      days: 0,
+      hours: 0,
+      min: 0,
+      sec: 0
+    }
+  }
+
+  componentDidMount() {
+    // update every second
+    this.interval = setInterval(() => {
+      const date = this.calculateCountdown(this.props.date)
+      date ? this.setState(date) : this.stop()
```

Comment:
This is correct, but I read date and got confused. Was about to point out that you where setting a date in `setState()`. This happens not only because of the name `date` itself, but because the prop `date` is a number

Maybe `const timeLeft = this.calculateCountdown(this.props.date)` (borrowing from calculateCountdown) or `const newTimeLeftState = this.calculateCountdown(this.props.date)`